### PR TITLE
[NEW-UI] Add task filter reducer

### PR DESCRIPTION
**Pull request by zankevich**

### CODE REVIEW

**Comment by bsekachev** [comment]
Can't run UI from this branch. 
```
npm install
npm start
```

There isn't any errors, server is started, all request to server in dev console are successful, but nothing happens, just white screen.

**Comment by zankevich** [comment]
> Can't run UI from this branch.
> 
> ```
> npm install
> npm start
> ```
> 
> There aren't any errors, server is started, all request to server in dev console are successful, but nothing happens, just white screen.
> Screenshot from 2019-07-24 19-06-15
> Screenshot from 2019-07-24 19-06-57

Any browser console errors?

**Comment by bsekachev** [comment]
@zankevich 
No any browser console errors

**Comment by zankevich** [comment]
> @zankevich
> No any browser console errors

OK, I see. I changed routing a little. Can you go to `/dashboard`. It should be fine then.

**Comment by bsekachev** [comment]
@zankevich 
Got it. Works

File: cvat-ui/src/components/app/app.tsx [hunk 1]
```diff
@@ -35,12 +26,15 @@ class App extends PureComponent<any, any> {
     return(
       <Router>
         <div>
-          <Redirect from="/" to="dashboard" />
```
Comment (bsekachev) [inline cvat-ui/src/components/app/app.tsx]: Why did you cancel redirection?

**Comment by bsekachev** [comment]
@zankevich 
Does a search by specific fields (name, status, etc) work here?

File: cvat-ui/src/components/app/app.tsx (hunk 1 above)
Comment (zankevich) [reply cvat-ui/src/components/app/app.tsx]: There are a couple of issues with this right now. I will return to this later to provide more robust approach for redirections.

**Comment by zankevich** [comment]
> @zankevich
> Does a search by specific fields (name, status, etc) work here?

`page` and `search` filters supported. Do I need to add something else?

**Comment by bsekachev** [comment]
@zankevich 
Production dashboard supports also such requests as

name: specific task name
mode: annotation/interpolation
status: annotation/validation/completed
owner: username
assignee: username

File: cvat-ui/src/components/dashboard/content/dashboard-content.tsx [hunk 2]
```diff
@@ -79,15 +81,13 @@ class DashboardContent extends Component<any, any> {
   }
 
   private onDeleteTask(task: any) {
```
Comment (bsekachev) [inline cvat-ui/src/components/dashboard/content/dashboard-content.tsx]: Why don't setup specific type (window.cvat.classes.Task)?

File: cvat-ui/src/components/dashboard/content/dashboard-content.tsx [hunk 3]
```diff
@@ -79,15 +81,13 @@ class DashboardContent extends Component<any, any> {
...
-    const props = this.props;
-
     confirm({
       title: 'Do you want to delete this task?',
       okText: 'Yes',
       okType: 'danger',
       centered: true,
       onOk() {
-        return props.deleteTask(task);
+        return task.delete();
```
Comment (bsekachev) [inline cvat-ui/src/components/dashboard/content/dashboard-content.tsx]: It is async function. What if promise is rejected? Will it be handled?

File: cvat-ui/src/components/dashboard/content/dashboard-content.tsx (hunk 2 above)
Comment (zankevich) [reply cvat-ui/src/components/dashboard/content/dashboard-content.tsx]: It should be declared on `cvat-core` side and then can be imported and used here. I can declare types on client side but this approach will work until something changed on `cvat-core`.

File: cvat-ui/src/components/dashboard/content/dashboard-content.tsx (hunk 3 above)
Comment (zankevich) [reply cvat-ui/src/components/dashboard/content/dashboard-content.tsx]: This is not a final solution. I'm working on this part right now.

**Comment by zankevich** [comment]
> @zankevich
> Production dashboard supports also such requests as
> 
> name: specific task name
> mode: annotation/interpolation
> status: annotation/validation/completed
> owner: username
> assignee: username

How to search exactly by name using search input on dashboard page?

File: cvat-ui/src/components/dashboard/content/dashboard-content.tsx (hunk 3 above)
Comment (bsekachev) [reply cvat-ui/src/components/dashboard/content/dashboard-content.tsx]: Could you please add TODO marker into the code?

File: cvat-ui/src/components/app/app.tsx (hunk 1 above)
Comment (bsekachev) [reply cvat-ui/src/components/app/app.tsx]: Could you please add TODO marker into the code?

**Comment by bsekachev** [comment]
> > @zankevich
> > Production dashboard supports also such requests as
> > name: specific task name
> > mode: annotation/interpolation
> > status: annotation/validation/completed
> > owner: username
> > assignee: username
> 
> How to search exactly by name using search input on dashboard page? I thought search input should use `search` filter.

In order to find by a name, in search field we enter: "name: <task name>". And the same with other fields.

**Pull request merged**